app/analysis.py

```diff
@@ -0,0 +1,23 @@
+import spacy
+from collections import Counter
+
+
+def nlp_analysis (user_responses):
+    """Returns the relevant topics for analysis usage
+    
+    takes in a list of (str)responses and performs nlp analysis
+    
+    Returns:
+    (dict) unordered dict of topics from mentee responses collection
+  
+    """
+
+    nlp = spacy.load("en_core_web_sm")
+   
+    responses_list=[]
+    for word in user_responses:
+        for token in nlp(word):
+            if not token.is_stop and not token.is_punct and token.pos_ in('ADJ','VERB','NOUN'):
```

Comment:
yup this is fantastic! If the word is an adjective, verb or noun it will never be a stop word or punctuation. So removing those checks is a great optimisation.

Reply:
Oh shoot haha so true. I hadn't considered that simplification.